python/ray/train/_internal/session.py

```diff
@@ -80,6 +80,31 @@ class TrainingResult:
     metadata: Optional[Dict] = None
 
 
+class _FutureTrainingResult:
```

Comment:
This wrapper is only used by PBT and it serves a similar purpose as the previous future in _TrackedCheckpoint.

It's not used anywhere else (except being created in the tune controller).